core/src/transaction_scheduler.rs

```diff
@@ -0,0 +1,100 @@
+//! Interface for a transaction scheduler to be used by the banking stage.
+//!
+
+use {
+    crate::{
+        bank_process_decision::BankPacketProcessingDecision,
+        immutable_deserialized_packet::ImmutableDeserializedPacket,
+    },
+    crossbeam_channel::RecvTimeoutError,
+    std::{rc::Rc, time::Duration},
+};
+
+pub mod priority_queue_scheduler;
+
+pub trait TransactionSchedulerBankingHandle {
+    /// Get the next batch of transactions to process, and the instructions for processing them.
+    fn get_next_transaction_batch(
+        &mut self,
+        timeout: Duration,
+    ) -> Result<Rc<ScheduledPacketBatch>, RecvTimeoutError>;
+
+    /// Signal that the current batch of transactions has been processed.
+    fn complete_batch(&mut self, batch: ProcessedPacketBatch);
+
+    /// Join thread(s) for scheduler if they exist
+    fn join(self) -> std::thread::Result<()>;
+}
+
+/// Message: [Scheduler -> BankingStage]
+/// The next batch of transactions to process, with instructions for processing them, and a unique id.
+pub struct ScheduledPacketBatch {
+    /// Unique identifier for the batch of transactions
+    pub id: ScheduledPacketBatchId,
+    /// Instruction for processing packets
+    pub processing_instruction: BankingProcessingInstruction,
+    /// Deserialized packets to process.
+    pub deserialized_packets: Vec<Rc<ImmutableDeserializedPacket>>,
```

Comment:
for account_lock aware schedulers we need `SanitizedTransaction`s, so we might want to modify this structure where we could send `SanitizedTransaction`s to tha bank for Consume, and packets for `Forward`. That way we don't need to sanitize in both the bank and scheduler.